### PR TITLE
Update circular-dependency.md

### PR DESCRIPTION
**Pull request by jakobwgnr**
was quite hard to find the issue here... Maybe it's good to add a warning about barrel files

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/nestjs/docs.nestjs.com/blob/master/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[x] Other... Please describe: Update Documentation
```

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?
N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information
N/A

### CODE REVIEW

**Comment by kamilmysliwiec** [comment]
Would you be able to give it a quick review? @jmcdo29 🔥

File: content/fundamentals/circular-dependency.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ While circular dependencies should be avoided where possible, you can't always d
 
 We also describe resolving circular dependencies between modules.
 
+> warning **Warning** A circular dependency might also be caused when using "barrel files"/index.ts files to group imports. Barrel files should be omitted when it comes to modules/providers. For more details please also see [this github issue](https://github.com/nestjs/nest/issues/1181#issuecomment-430197191).
```
Comment (jmcdo29) [inline content/fundamentals/circular-dependency.md:9]: I think we could re-word this to something along the lines of "Barrel files should not be used when importing files within the same directory as the barrel file, i.e. `cats/cats.controller` should not import `cats` to import the `cats/cats.service` file".

**Review by jmcdo29 — CHANGES_REQUESTED** [review]
Left a minor comment. I think it re-iterates Kamil's point in the last comment of the linked issue

File: content/fundamentals/circular-dependency.md (hunk 1 above)
Comment (jakobwgnr) [reply content/fundamentals/circular-dependency.md]: @jmcdo29 Thanks for the review. I have updated the text as suggested.

**Review by jmcdo29 — APPROVED** [review]

File: content/fundamentals/circular-dependency.md [hunk 2]
```diff
@@ -6,6 +6,8 @@ While circular dependencies should be avoided where possible, you can't always d
 
 We also describe resolving circular dependencies between modules.
 
+> warning **Warning** A circular dependency might also be caused when using "barrel files"/index.ts files to group imports. Barrel files should be omitted when it comes to modules/providers. So barrel files should not be used when importing files within the same directory as the barrel file, i.e. `cats/cats.controller` should not import `cats` to import the `cats/cats.service` file. For more details please also see [this github issue](https://github.com/nestjs/nest/issues/1181#issuecomment-430197191).
```
Comment (kamilmysliwiec) [inline content/fundamentals/circular-dependency.md:9]:
```suggestion
> warning **Warning** A circular dependency might also be caused when using "barrel files"/index.ts files to group imports. Barrel files should be omitted when it comes to module/provider classes. For example, barrel files should not be used when importing files within the same directory as the barrel file, i.e. `cats/cats.controller` should not import `cats` to import the `cats/cats.service` file. For more details please also see [this github issue](https://github.com/nestjs/nest/issues/1181#issuecomment-430197191). 
```

**Comment by kamilmysliwiec** [comment]
Thanks!

**Pull request merged**